src/authentication-strategies/JWT.strategy.ts

```diff
@@ -0,0 +1,34 @@
+const jwt = require('jsonwebtoken');
+import {promisify} from 'util';
+const verifyAsync = promisify(jwt.verify);
+// Consider turn it to a binding
+const SECRET = 'secretforjwt';
+import {Request, HttpErrors} from '@loopback/rest';
+import {UserProfile} from '@loopback/authentication';
+import * as _ from 'lodash';
+import {AuthenticationStrategy} from './authentication.strategy';
+
+export class JWTStrategy implements AuthenticationStrategy {
+  async authenticate(request: Request): Promise<UserProfile | undefined> {
+    let token = request.query.access_token || request.headers['authorization'];
+    if (!token) throw new HttpErrors.Unauthorized('No access token found!');
+
+    if (token.startsWith('Bearer ')) {
+      token = token.slice(7, token.length);
+    }
+
+    try {
+      const decoded = await verifyAsync(token, SECRET);
```

Comment:
Based on our discussion around `login`, I feel this place should be changed in a similar way too.

- JWTStrategy deals with REST-specific aspects only: how to extract the token from the request, how to bind UserProfile as request's current user.
- Validation of the token and conversion into UserProfile should be handled by a new utility/service. This can be possibly the same class/file where `login` is implemented.

I expect the same benefits:
- Other transports like gRPC and WebSocket can use the same utility/service to validate access tokens and convert them into user profiles.
- The utility/service can be tested in isolation with no transport involved.

Reply:
@bajtos I removed the strategy written by us, replaced with the Strategy in `passport-jwt`, which is one of tens passport strategies that works with `passport` module out-of-box.

Reply:
> Based on our discussion around `login`, I feel this place should be changed in a similar way too.
>
> - JWTStrategy deals with REST-specific aspects only: how to extract the token from the request, how to bind UserProfile as request's current user.
> - Validation of the token and conversion into UserProfile should be handled by a new utility/service. This can be possibly the same class/file where `login` is implemented.

@jannyHou I feel this comment should be addressed before landing the patch. If it makes it easier for you, then I am ok to defer this work to a follow-up pull request, as long as the change is made as part of the actual user story this pull request belongs to.